docs/iris/src/userguide/real_and_lazy_data.rst

```diff
@@ -0,0 +1,246 @@
+.. _real_and_lazy_data:
+
+==================
+Real and Lazy Data
+==================
+
+What is Real and Lazy Data?
+---------------------------
+
+Every Iris cube contains an n-dimensional data array, which could be real or
+lazy.
+
+Real data is contained in a NumPy array which has a shape, a data type, some
+other useful information and many data points, each of which use up a small
+allocation of memory.
+
+Lazy data is contained in a conceptual array which retains the information
+about its real counterpart but has no actual data points, so its memory
+allocation is much smaller.  This will be in the form of a Dask array.
+
+Arrays in Iris can be converted between their real and lazy states,
+although there are some limits to this process (*explain this better - what
+limits?*).
+
+The advantage of using lazy data is that it has a small memory footprint which
+enables the user to load and manipulate datasets that you would otherwise not
+be able to fit into memory.
+
+However, in order to execute certain operations (such as calculations on actual
+data values) the real data must be realized.  Using Dask, the operation will
+be deferred until you request the result, at which point it will be executed
+using Dask's parallel processing schedulers.  The combination of these two
+behaviours can offer a significant performance boost.
+
+You can check whether the data array on your cube is lazy using the Iris
+function 'has_lazy_data'.  For example:
+
+.. doctest::
+
+    >>> import iris
+    >>> cube = iris.load_cube(filename, 'air_temp.pp')
+    >>> cube.has_lazy_data()
+    True
+    >>> cube.data
+    >>> cube.has_lazy_data()
+    False
+
+When does my Data Become Real?
+------------------------------
+
+If the data on your cube is in its lazy state, it will only become real if you
+'touch' the data.  This means any way of directly accessing the data, such as
+assigning it to a variable or simply using 'cube.data' as in the example above.
+
+Any action which requires the use of actual data values (such as cube maths)
+will also cause the data to be loaded into memory, although data realization
+is always deferred until the result is requested:
+
+.. doctest::
+
+    >>> cube = iris.load_cube(iris.sample_data_path('air_temp.pp'))
+    >>> cube.has_lazy_data()
+    True
+    >>> cube += 5
+    >>> cube.has_lazy_data()
+    True
+    >>> cube.data
+    >>> cube.has_lazy_data()
+    False
+
+You can also convert realized data back into a lazy array:
+
+.. doctest::
+
+    >>> cube.has_lazy_data()
+    False
+    >>> cube.data = cube.lazy_data()
+    >>> cube.has_lazy_data()
+    True
+
+Core data refers to the current state of the cube's data, be it real or
+lazy.  This can be used if you wish to refer to the data array but are
+indifferent to its current state.  If the cube's data is lazy, it will not be
```

Comment:
Again, don't much like this use of "state".
Maybe "This can be used when you want to use the data array in some way that works regardless of whether it is real or lazy, such as indexing it".